oslab/oslab2_memory_management/oslab2.1_memory_management_simulation/config/config.txt
1000
354 238 239 965 245 601 698 673 561 708 128 394 353 539 413 191 668 786 370 430 411 26 627 475 215 614 262 168 671 847 544 533 816 394 83 478 562 700 565 1000 801 397 607 704 756 176 169 554 491 802 316 490 490 308 420 147 951 919 908 400 133 24 387 744 688 338 151 812 726 677 616 249 151 417 208 979 72 820 541 396 241 58 911 397 456 476 68 898 806 928 136 772 708 469 388 98 527 61 251 606 886 558 978 136 795 898 103 406 54 808 764 771 768 836 618 556 384 241 365 595 41 871 480 679 188 145 521 628 157 323 496 634 455 888 265 551 68 270 876 440 292 354 846 672 806 545 738 160 352 967 1 222 260 36 663 341 836 361 127 424 279 726 476 617 527 251 241 423 989 250 597 230 638 176 94 964 552 106 559 452 879 934 256 163 814 449 83 961 618 160 790 763 144 512 480 771 503 775 978 673 99 629 621 702 821 993 179 967 756 616 94 640 117 293 521 535 247 110 931 979 345 252 79 395 971 679 756 441 618 196 55 420 890 795 635 723 970 923 945 964 500 757 772 524 289 28 292 833 750 113 708 965 999 405 419 708 223 173 732 209 762 361 607 925 320 514 752 0 363 412 981 562 849 133 748 246 952 49 494 606 87 541 488 129 661 425 159 232 398 396 419 210 911 588 489 454 94 540 989 704 513 487 7 312 525 705 32 257 978 813 774 229 651 243 707 855 92 7 133 496 266 532 632 627 834 691 254 578 427 522 423 684 918 218 716 673 238 642 475 546 450 121 794 543 601 731 897 343 742 65 84 211 700 310 37 124 874 218 601 491 586 291 370 967 174 420 109 240 77 60 424 480 913 681 837 114 181 66 401 795 898 649 410 481 185 794 864 853 587 434 562 848 514 855 635 431 216 350 365 478 901 211 293 101 888 713 423 133 188 880 903 41 161 446 846 374 401 438 559 981 230 84 627 681 879 973 965 91 657 745 727 900 82 858 432 186 902 389 94 80 699 326 40 812 157 581 885 769 667 699 523 799 994 121 105 127 34 515 467 390 37 400 773 278 934 960 636 853 94 492 425 594 866 709 980 566 245 783 398 691 181 714 847 333 273 229 33 159 432 276 913 225 238 785 512 753 558 248 533 276 493 638 946 836 758 343 54 917 700 619 543 629 356 487 694 720 968 659 454 719 607 920 235 26 739 128 160 563 31 708 638 93 837 42 38 995 56 63 219 336 258 477 955 75 601 275 745 149 946 27 600 631 994 866 86 534 944 151 726 685 602 503 746 460 638 730 553 955 824 44 822 862 728 780 822 376 154 869 570 740 810 490 449 174 686 454 561 666 98 312 975 253 380 142 657 624 81 754 843 379 170 8 851 233 707 819 619 247 57 676 865 41 36 145 305 964 647 272 753 947 396 748 547 691 544 729 122 563 171 716 462 315 822 14 156 336 231 209 617 689 719 825 126 408 527 690 179 169 900 12 175 755 559 67 448 159 14 699 942 822 746 768 983 118 258 656 645 516 663 200 518 270 157 148 862 73 848 136 16 474 789 103 345 778 354 538 592 726 896 669 78 401 358 38 441 252 870 542 429 914 97 592 938 430 468 151 172 340 309 343 165 957 168 198 583 737 296 838 69 209 21 271 581 895 126 503 515 412 645 556 295 830 973 136 619 233 503 831 180 264 118 718 99 13 92 115 170 263 519 880 690 602 982 699 656 392 874 502 994 189 61 728 506 537 339 378 263 678 446 917 991 435 926 66 334 967 586 72 688 992 907 44 380 820 139 993 246 537 278 236 746 524 32 777 989 290 559 133 233 603 222 890 954 451 133 389 382 503 283 785 935 668 459 450 173 926 140 653 260 14 635 888 669 947 402 911 439 936 367 108 998 806 792 671 723 461 314 216 994 991 102 172 628 352 227 561 646 362 973 375 14 441 170 281 797 327 635 931 306 530 204 949 489 616 968 983 104 713 380 1 573 817 439 924 752 109 483 414 937 459 16 839 351 638 969 279 384 920 861 297 702 791 964 369 445 195 897 551 393 109 944 836 400 224 904 268 547 368 147 464 932 278 710 667 497 980 208 154 980 160 920 339 488 920 661 374 537 221 506 312 791 938 966 892 505 464 752 864 475 666 605 155 937 113 352 45 316 164 248 524 259 390 908 316 152 577 357 753 290 49 717 265 413 679 958 299 230 675 951 949 280 75 562 502 724 923 312 795 518 281 474 125 724 632 739 355 964 888 916 187 489 898 193 134 497 730 194 54 546 947 
1000
426 775 461 14 143 671 114 975 581 919 930 506 648 877 996 722 107 460 910 70 68 306 349 51 936 470 37 165 844 113 832 249 611 225 179 551 617 913 805 665 526 230 243 806 38 195 566 226 342 957 889 581 96 303 493 421 190 958 484 857 861 338 121 240 463 351 408 75 278 918 548 0 119 470 499 993 830 659 998 73 292 972 855 629 535 401 60 726 953 584 370 345 54 97 76 125 428 433 43 261 239 811 989 107 350 24 310 242 22 140 4 267 802 305 94 664 418 499 518 831 83 781 258 102 707 605 731 196 98 33 939 716 817 83 758 155 492 241 960 46 381 48 136 467 87 84 267 925 283 52 875 693 928 553 672 483 193 405 378 318 745 154 245 647 829 709 252 766 342 404 952 750 387 89 552 496 600 902 114 424 955 245 67 889 130 254 97 822 779 19 500 565 258 397 325 166 255 550 938 73 104 108 456 461 221 904 676 675 10 537 462 297 486 834 51 553 278 698 16 200 480 374 415 664 216 732 716 476 511 774 504 328 319 352 483 411 240 345 101 371 664 39 876 384 736 87 723 257 968 283 639 368 732 5 925 229 187 942 602 216 891 499 188 437 790 82 338 652 973 946 905 782 28 158 992 331 580 19 821 348 926 418 487 874 216 44 228 530 87 326 290 611 477 620 421 634 298 706 2 173 97 4 304 301 653 539 465 441 337 856 26 647 594 961 951 439 197 200 558 646 498 815 120 580 739 936 423 581 537 606 35 459 992 663 242 485 182 1000 262 259 442 333 917 404 581 730 638 595 973 630 751 338 24 158 361 61 374 394 544 562 561 977 21 308 573 220 361 190 964 116 886 910 23 401 848 768 179 45 2 429 686 842 854 837 485 122 354 870 1 742 558 337 418 630 803 731 548 829 497 850 921 244 168 263 568 115 117 634 736 461 48 231 115 61 862 890 88 164 473 982 331 482 153 732 369 610 325 810 468 259 645 645 587 389 428 897 919 860 164 119 970 722 782 451 125 678 396 211 294 973 324 118 834 532 835 872 30 12 894 683 105 985 439 679 881 591 266 312 431 474 718 438 120 522 713 834 931 568 490 421 660 909 238 31 387 864 166 937 325 765 647 312 708 634 581 623 899 190 418 771 772 865 833 291 768 641 980 737 203 840 950 924 592 223 197 816 40 875 353 244 137 594 456 938 742 877 733 85 936 534 898 690 297 171 600 368 641 914 504 101 252 496 751 568 481 701 851 475 931 520 649 210 737 551 503 910 869 484 412 400 18 737 189 467 650 397 972 793 818 560 485 710 736 67 712 783 675 81 387 38 726 605 556 393 135 493 658 483 818 458 170 514 233 640 299 492 284 501 611 117 432 436 778 826 752 581 711 973 40 60 187 356 482 598 257 239 119 727 605 309 21 175 45 264 179 318 985 352 254 640 648 275 624 35 36 57 851 753 898 892 374 629 688 18 446 188 584 990 237 264 223 483 117 314 571 937 80 217 587 166 794 788 353 912 276 565 918 291 59 306 5 640 700 349 488 236 803 547 462 155 260 130 920 102 557 340 407 834 826 731 429 854 474 640 943 884 375 54 673 154 576 778 69 610 872 392 824 626 84 890 316 528 83 990 670 1000 183 912 292 449 805 844 175 112 803 941 220 374 569 852 953 908 205 245 170 991 732 150 755 225 684 65 596 846 811 334 894 181 260 716 77 297 979 852 459 254 354 138 899 906 915 954 748 420 17 171 792 382 964 282 327 813 586 957 628 917 86 389 20 812 864 780 700 427 347 943 44 326 820 983 964 64 829 33 646 107 45 672 653 190 738 215 187 37 810 485 409 389 43 112 203 678 192 970 826 647 739 303 543 751 307 797 683 924 195 194 912 938 733 580 915 232 944 881 570 414 789 463 610 438 911 192 388 70 359 9 565 32 491 30 41 337 763 496 470 49 23 680 551 922 898 596 656 694 292 604 366 832 319 821 933 29 998 568 263 810 200 782 505 11 684 191 12 367 550 291 219 724 623 642 311 173 717 416 763 952 6 572 95 513 79 213 991 64 88 534 554 618 837 697 637 465 447 557 251 862 282 812 120 721 928 160 92 54 397 273 675 571 532 313 828 645 780 986 903 816 882 310 22 362 954 144 288 213 247 940 872 148 511 700 182 201 918 956 193 190 893 39 124 91 192 738 42 991 871 989 353 974 924 446 670 111 101 447 648 646 907 37 822 630 889 185 342 363 937 743 136 596 196 790 915 346 177 346 416 129 189 871 873 786 

oslab/oslab2_memory_management/oslab2.1_memory_management_simulation/config/in_frag_2_config.txt
1000
803 56 704 255 356 966 746 486 719 496 37 891 976 242 122 839 387 976 674 602 55 473 159 209 737 165 6 841 924 733 47 479 563 134 64 19 582 204 958 856 608 156 617 222 94 848 389 781 439 978 470 224 717 773 556 228 449 601 290 263 419 399 603 730 364 700 239 161 687 274 783 379 620 768 389 815 657 304 495 159 538 881 813 783 246 605 541 941 19 585 127 522 688 970 85 863 37 573 167 898 147 825 57 854 431 134 31 397 769 166 301 17 794 655 798 917 993 620 530 666 499 593 11 914 235 491 477 105 444 812 261 169 141 364 608 272 18 144 841 651 556 95 842 970 576 723 233 575 639 51 568 753 707 618 908 691 22 696 175 611 545 917 170 826 881 579 269 130 835 388 613 721 833 700 628 246 8 48 231 831 851 135 27 424 926 792 6 913 8 231 108 743 94 770 298 561 63 639 134 126 377 913 282 445 316 67 362 399 635 957 183 421 230 141 933 116 477 837 692 139 89 136 736 649 387 848 994 483 567 274 284 613 722 626 400 171 76 59 694 198 276 974 388 435 817 424 649 959 180 70 662 283 526 58 973 773 770 981 578 416 566 28 645 236 772 874 772 105 69 691 558 362 790 8 798 813 880 193 334 460 800 136 250 630 714 105 55 755 36 634 751 379 723 71 384 623 365 767 727 555 944 713 357 40 700 565 533 98 120 306 597 291 594 793 302 708 695 780 58 912 685 907 930 220 193 91 155 221 563 979 223 319 902 883 802 719 827 436 347 557 927 467 391 644 661 732 147 280 560 4 158 498 334 880 829 554 860 69 792 8 44 812 190 419 97 754 617 57 330 43 769 71 306 318 487 385 441 637 53 966 996 898 728 545 932 486 355 689 717 851 325 194 260 713 52 847 698 388 727 807 974 697 617 324 798 701 306 311 920 40 109 190 897 998 819 66 238 875 391 635 344 726 361 305 71 133 461 665 427 87 577 658 928 908 349 45 818 519 733 541 124 63 326 479 477 814 738 349 639 691 382 661 811 637 354 875 47 236 619 165 964 830 797 97 324 497 333 290 922 884 892 468 179 637 244 683 456 427 143 828 625 414 757 281 67 306 632 667 879 799 357 188 803 321 717 254 244 1000 295 954 176 855 781 966 695 83 639 421 669 370 885 435 650 486 109 494 396 16 970 105 398 815 75 33 580 913 547 831 656 177 26 650 167 967 589 319 844 504 547 629 475 832 641 359 963 375 23 776 432 622 641 350 272 941 896 258 177 388 534 161 761 749 543 708 742 107 715 889 74 594 639 94 405 115 574 541 62 544 922 822 563 140 245 300 991 296 485 280 81 778 526 256 123 115 77 336 589 398 766 974 914 845 359 65 75 148 388 217 813 7 308 750 200 505 237 472 26 809 477 905 881 165 115 721 956 680 333 914 540 736 821 332 244 260 445 604 304 471 6 646 907 652 998 188 311 516 113 113 499 94 110 228 454 441 141 584 261 813 685 764 311 196 150 514 184 144 159 657 602 562 83 270 734 328 713 807 880 792 204 343 66 662 209 724 915 568 450 912 320 254 779 936 859 733 307 388 163 722 588 725 222 94 205 835 942 472 357 264 523 301 766 249 566 836 58 849 707 433 553 764 883 834 430 295 989 199 870 403 204 527 483 697 398 224 956 231 679 39 481 206 547 663 389 63 919 999 655 486 196 941 690 912 742 834 315 870 249 930 618 13 192 823 681 94 826 944 534 722 910 474 774 475 717 869 320 636 986 140 181 299 91 568 379 356 587 63 687 308 162 63 178 568 171 427 442 929 174 901 564 959 268 83 360 61 540 435 238 572 332 700 574 873 547 756 650 623 20 304 150 367 506 337 996 107 90 814 49 143 193 478 259 765 282 277 45 961 422 46 312 937 341 294 124 25 739 331 255 539 306 660 643 602 471 187 913 187 1000 434 408 304 792 108 973 397 183 3 776 253 68 32 558 382 155 264 377 766 632 91 523 786 78 384 615 367 270 995 448 525 755 783 613 338 126 868 775 656 996 853 822 423 638 40 744 108 274 91 154 496 832 264 383 647 759 581 946 257 297 693 265 465 355 728 963 150 329 215 483 70 463 522 567 695 927 523 524 408 815 6 218 702 169 507 573 306 933 101 623 553 809 737 916 777 159 637 406 709 660 386 287 407 758 755 366 64 71 401 703 217 489 405 728 228 984 687 74 781 929 724 847 623 128 153 284 264 94 498 373 715 939 111 460 131 217 105 8 604 95 343 611 555 
100
791 56 44 699 965 531 618 484 603 477 699 842 791 874 294 117 485 214 284 213 655 810 155 56 246 493 804 401 357 44 151 654 462 440 666 460 756 393 469 46 986 770 42 133 758 810 158 95 139 107 611 257 602 176 924 276 380 420 972 973 936 294 929 591 120 106 7 756 954 717 112 867 201 566 378 421 343 89 720 320 167 143 588 532 834 612 655 150 276 10 907 960 459 737 614 650 664 12 662 983 


python/final/lianjia/settings.py
import random

# Scrapy settings for lianjia project
#
# For simplicity, this file contains only settings considered important or
# commonly used. You can find more settings consulting the documentation:
#
#     https://docs.scrapy.org/en/latest/topics/settings.html
#     https://docs.scrapy.org/en/latest/topics/downloader-middleware.html
#     https://docs.scrapy.org/en/latest/topics/spider-middleware.html

BOT_NAME = "lianjia"

SPIDER_MODULES = ["lianjia.spiders"]
NEWSPIDER_MODULE = "lianjia.spiders"


# Crawl responsibly by identifying yourself (and your website) on the user-agent
USER_AGENT = "user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.82 Safari/537.36"

# Obey robots.txt rules
ROBOTSTXT_OBEY = False

# Configure maximum concurrent requests performed by Scrapy (default: 16)
# CONCURRENT_REQUESTS = 32

# Configure a delay for requests for the same website (default: 0)
# See https://docs.scrapy.org/en/latest/topics/settings.html#download-delay
# See also autothrottle settings and docs
# Random download delay between 3 and 10 seconds.
DOWNLOAD_DELAY = random.randint(3, 10)
# The download delay setting will honor only one of:
CONCURRENT_REQUESTS_PER_DOMAIN = 2
CONCURRENT_REQUESTS_PER_IP = 2

# Disable cookies (enabled by default)
COOKIES_ENABLED = False

# Disable Telnet Console (enabled by default)
# TELNETCONSOLE_ENABLED = False

# Override the default request headers:
DEFAULT_REQUEST_HEADERS = {
    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
    "Accept-Language": "en",
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36",
}

# Enable or disable spider middlewares
# See https://docs.scrapy.org/en/latest/topics/spider-middleware.html
# SPIDER_MIDDLEWARES = {
#    'lianjia.middlewares.LianjiaSpiderMiddleware': 543,
# }

# Enable or disable downloader middlewares
# See https://docs.scrapy.org/en/latest/topics/downloader-middleware.html
# Disable scrapy's default UserAgentMiddleware to use RandomUserAgent
# Enable scrapy proxy
DOWNLOADER_MIDDLEWARES = {
    # "scrapy.downloadermiddlewares.useragent.UserAgentMiddleware": None,
    # "scrapy_user_agents.middlewares.RandomUserAgentMiddleware": 400,
    # "lianjia.middlewares.ProxiesMiddleware": 300,
}

# Enable or disable extensions
# See https://docs.scrapy.org/en/latest/topics/extensions.html
# EXTENSIONS = {
#    'scrapy.extensions.telnet.TelnetConsole': None,
# }

# Configure item pipelines
# See https://docs.scrapy.org/en/latest/topics/item-pipeline.html
# ITEM_PIPELINES = {
#    'lianjia.pipelines.LianjiaPipeline': 300,
# }

# Enable and configure the AutoThrottle extension (disabled by default)
# See https://docs.scrapy.org/en/latest/topics/autothrottle.html
# AUTOTHROTTLE_ENABLED = True
# The initial download delay
# AUTOTHROTTLE_START_DELAY = 5
# The maximum download delay to be set in case of high latencies
# AUTOTHROTTLE_MAX_DELAY = 60
# The average number of requests Scrapy should be sending in parallel to
# each remote server
# AUTOTHROTTLE_TARGET_CONCURRENCY = 1.0
# Enable showing throttling stats for every response received:
# AUTOTHROTTLE_DEBUG = False

# Enable and configure HTTP caching (disabled by default)
# See https://docs.scrapy.org/en/latest/topics/downloader-middleware.html#httpcache-middleware-settings
# HTTPCACHE_ENABLED = True
# HTTPCACHE_EXPIRATION_SECS = 0
# HTTPCACHE_DIR = 'httpcache'
# HTTPCACHE_IGNORE_HTTP_CODES = []
# HTTPCACHE_STORAGE = 'scrapy.extensions.httpcache.FilesystemCacheStorage'

# Set settings whose default value is deprecated to a future-proof value
REQUEST_FINGERPRINTER_IMPLEMENTATION = "2.7"
TWISTED_REACTOR = "twisted.internet.asyncioreactor.AsyncioSelectorReactor"
